### PR TITLE
Feature/button tests

### PR DESCRIPTION
**Pull request by MKaHead**
testcafe:
- Introduce name convention for UI test files -> `ui.test.js` for unit test we could with `unit.test.js` 

typo mixin:
- Remove obsolete font weights variables, and add relevant ones

Button improvements:
- Add typo mixin
- Improve button click callback function 

Button Test/Storys/Demo:
- Add UI tests for Button
- Improve story for click callback function 
- Improve react button demo 

ESlint:
- Add fixture and test as global variables
- Add exception rule for import/no-extraneous-dependencies rule for `ui.test.js ` files

package.json
- Update Storybook from 5.0.3 to 5.0.5

### CODE REVIEW

File: src/components/10-atoms/button/test.ui.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  const $axaButtonShadow = await Selector(() => document.querySelector('axa-button').shadowRoot);
+  const $axaButtonShadowEl = await $axaButtonShadow.find(BUTTON_CLASS);
+  await t.expect(await $axaButtonShadowEl.hasAttribute('disabled')).ok();
+});
+
+test('should set button element type', async t => {
+  const setType = ClientFunction( () => {
+    document.querySelector('axa-button').type = 'submit';
+  });
+  await setType();
```
Comment (raphaellueckl) [inline src/components/10-atoms/button/test.ui.js]: cool!

File: src/components/10-atoms/button/test.ui.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import { Selector, ClientFunction } from 'testcafe';
+
+fixture('Button - basic functionality').page('http://localhost:9999/iframe.html?id=atoms-button--button-click-event');
+
+const BUTTON_TAG = 'axa-button';
+const BUTTON_CLASS = '.a-button';
+
+test('should render button', async t => {
```
Comment (raphaellueckl) [inline src/components/10-atoms/button/test.ui.js]: We should decide on upper or lowercase "S" for "should".

File: src/demo/react/DemoButton.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React, { createElement } from 'react';
+import '../../components/10-atoms/button';
+import createAXAButtonReact from '../../components/10-atoms/button/index.react';
+
+const AXAButtonReact = createAXAButtonReact(createElement);
+
+const DemoButton = () => {
+  const handleAXAButtonClick = () => {
+    /* eslint-disable no-alert */
+    window.alert('Oo yeah you clicked me');
```
Comment (raphaellueckl) [inline src/demo/react/DemoButton.jsx]: Can't we workaround this ugly eslint rule somehow? Like can we use something else than alert or completely remove this rule from eslint (or ignore ".story.js" files)?

Comment (raphaellueckl) [reply src/demo/react/DemoButton.jsx]: #6e36fa2

Comment (raphaellueckl) [reply src/demo/react/DemoButton.jsx]: 6e36fa2622bcb1a6cda401d9535aee5d8ec02249

Comment (raphaellueckl) [reply src/demo/react/DemoButton.jsx]: 6e36fa2

Comment (MKaHead) [reply src/demo/react/DemoButton.jsx]: Improve example without a alert see 6e36fa2

**Comment by raphaellueckl** [comment]
Do those tests run on browserstack with IE?

**Pull request merged**